shanghai/connection.py

```diff
@@ -24,6 +24,9 @@ def writeline(self, line: bytes):
         self.writer.write(line)
         self.writer.write(b'\r\n')
 
+    def force_close(self):
+        self.writer.close()
+
     async def close(self):
         await self.writer.drain()
```

Comment:
`BaseTransport.close(self)`
> Close the transport. If the transport has a buffer for outgoing data, buffered data will be flushed asynchronously.

Meaning, we can ditch `.drain()` and force_close and close become equivalent.